modules/switch/react/spec/Switch.spec.tsx

```diff
@@ -1,44 +1,131 @@
+      expect(getByRole(role)).toHaveProperty('type', 'checkbox');
+    });
+
+    it('should be unchecked by default', () => {
+      const {getByRole} = render(<Switch onChange={cb} />);
+      expect(getByRole(role)).toHaveProperty('checked', false);
+    });
+
+    test('should have a "pointer" cursor', () => {
+      const {getByRole} = render(<Switch onChange={cb} />);
+      expect(getByRole(role)).toHaveStyleRule('cursor', 'pointer');
+    });
+  });
+
+  describe('when rendered with checked=true', () => {
+    it('should render a checked checkbox', () => {
+      const {getByRole} = render(<Switch checked={true} onChange={cb} />);
+      expect(getByRole(role)).toHaveProperty('checked', true);
+    });
+  });
+
+  describe('when rendered with an id', () => {
+    it('should render a checkbox with that id', () => {
+      const id = 'switchy';
+      const {getByRole} = render(<Switch id={id} onChange={cb} />);
+
+      expect(getByRole(role)).toHaveAttribute('id', id);
+    });
+  });
+
+  describe('when rendered without an id', () => {
+    it('should create a unique id for each instance', () => {
+      const testIds = ['test1', 'test2'];
+      const {getByTestId} = render(
+        <>
+          <Switch data-testid={testIds[0]} onChange={cb} />
+          <Switch data-testid={testIds[1]} onChange={cb} />
+        </>
+      );
+
+      const id1 = getByTestId(testIds[0]).getAttribute('id');
+      const id2 = getByTestId(testIds[1]).getAttribute('id');
+
+      expect(id1).not.toEqual(id2);
+    });
+
+    it('should keep the same unique id if re-rendered', () => {
+      const {getByRole, rerender} = render(<Switch checked={false} onChange={cb} />);
+
+      const uniqueId = getByRole(role).getAttribute('id');
+      expect(getByRole(role)).toHaveProperty('id', uniqueId);
+
+      rerender(<Switch checked={true} onChange={cb} />);
+
+      expect(getByRole(role)).toHaveProperty('checked');
+      expect(getByRole(role)).toHaveProperty('id', uniqueId);
+    });
+  });
+
+  describe('when rendered with the disabled prop', () => {
+    it('should render a disabled checkbox', () => {
+      const {getByRole} = render(<Switch disabled={true} onChange={cb} />);
+      expect(getByRole(role)).toHaveProperty('disabled', true);
+    });
+    it('should have a "not-allowed" cursor', () => {
+      const {getByRole} = render(<Switch disabled={true} onChange={cb} />);
+      expect(getByRole(role)).toHaveStyleRule('cursor', 'not-allowed');
+    });
+  });
+
+  describe('when rendered with extra, arbitrary props', () => {
+    it('should apply those extra props to the checkbox', () => {
+      const testValue = 'test';
+
+      const {getByRole} = render(<Switch data-propspread={testValue} onChange={cb} />);
+      expect(getByRole(role)).toHaveAttribute('data-propspread', testValue);
+    });
+  });
+
+  describe('when rendered with an input ref', () => {
+    it('the current property should not be null', () => {
```

Comment:
I find it odd when these don't read as a sentence (i.e. "when rendered with an input ref, it the current...")

What if we changed this to 
```suggestion
    it('should set the ref current property', () => {
```

I also think we could combine these two tests and just have two `expect`s, with a name like `should set the ref to the checkbox input element`

Reply:
Yea I like that, it was just reading funny from a test output. I'll make one minor adjustment to make the `ref` possessive`